ray-operator/controllers/ray/rayjob_controller.go

```diff
@@ -333,9 +335,16 @@ func (r *RayJobReconciler) Reconcile(ctx context.Context, request ctrl.Request)
 				logger.Info(fmt.Sprintf("shutdownTime not reached, requeue this RayJob for %d seconds", delta))
 				return ctrl.Result{RequeueAfter: time.Duration(delta) * time.Second}, nil
 			}
-			// We only need to delete the RayCluster. We don't need to delete the submitter Kubernetes Job so that users can still access
-			// the driver logs. In addition, a completed Kubernetes Job does not actually use any compute resources.
-			if _, err = r.deleteClusterResources(ctx, rayJobInstance); err != nil {
+			if os.Getenv(DELETE_RAYJOB_CR_AFTER_JOB_FINISHES) == "true" {
+				err = r.Client.Delete(ctx, rayJobInstance)
+				logger.Info("RayJob CR is deleted", "RayJob", rayJobInstance.Name)
+			} else {
+				// We only need to delete the RayCluster. We don't need to delete the submitter Kubernetes Job so that users can still access
+				// the driver logs. In addition, a completed Kubernetes Job does not actually use any compute resources.
+				_, err = r.deleteClusterResources(ctx, rayJobInstance)
+				logger.Info("RayCluster is deleted", "RayJob", rayJobInstance.Name, "RayCluster", rayJobInstance.Status.RayClusterName)
```

Comment:
Check whether RayJob CR name and RayCluster CR name are redundant or not.

Reply:
You're right. The RayJob CR name is redundant. But the RayCluster CR name is not redundant. I've deleted the RayJob CR name. Now the log looks like this:

```
{"level":"info","ts":"2024-07-10T17:30:37.191+0800","logger":"controllers.RayJob","msg":"RayCluster is deleted","RayJob":{"name":"rayjob-sample","namespace":"default"},"reconcileID":"4ea84c67-e4c1-4030-bed7-c8f5eecbb647","RayCluster":"rayjob-sample-raycluster-r55c6"}
```